pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/LLCSegmentCommit.java

```diff
@@ -109,22 +110,27 @@ boolean uploadSegment(final String instanceId, final String segmentNameStr) {
     final List<FileItem> items;
 
     try {
-      // The following statement blocks until the entire segment is read into memory.
+      // The following statement blocks until the entire segment is read into memory/disk.
       items = upload.parseRequest(getRequest());
 
-      boolean found = false;
       File dataFile = null;
 
-      for (final Iterator<FileItem> it = items.iterator(); it.hasNext() && !found; ) {
-        final FileItem fi = it.next();
-        if (fi.getFieldName() != null && fi.getFieldName().equals(segmentNameStr)) {
-          found = true;
-          dataFile = new File(tempDir, segmentNameStr);
-          fi.write(dataFile);
+      for (FileItem fileItem : items) {
```

Comment:
Refactor this to use a single code path in a different PR. All of these are doing roughly the same thing, it would be nice to avoid duplicated code. :)